gui/wxpython/dbmgr/base.py

```diff
@@ -3856,10 +3863,10 @@ def OnChangeLayer(self, event):
         """Layer number of layer to be deleted is changed"""
         try:
             layer = int(event.GetString())
-        except:
+        except ValueError:
             try:
                 layer = self.mapDBInfo.layers.keys()[0]
-            except:
+            except KeyError:
```

Comment:
Assuming layers is a dictionary, this would always through out 
```
TypeError: 'dict_keys' object is not subscriptable
```
This may be some python 2 leftover. So perhaps actually making the keys into list is needed and then you need IndexError.
Or did I misunderstand this?

Reply:
Yeah, you're right - this would work in Python2 but not Python3. I'll adjust this